src/core.js

```diff
@@ -0,0 +1,661 @@
+
+    // Unsupported, non-API function.
+    youme.read.polyAftertouch = function (channel, data) {
+        return {
+            type: "aftertouch",
+            channel: channel,
+            note: data[1],
+            pressure: data[2]
+        };
+    };
+
+    // Unsupported, non-API function.
+    youme.read.controlChange = function (channel, data) {
+        return {
+            type: "control",
+            channel: channel,
+            number: data[1],
+            value: data[2]
+        };
+    };
+
+    // Unsupported, non-API function.
+    youme.read.programChange = function (channel, data) {
+        return {
+            type: "program",
+            channel: channel,
+            program: data[1]
+        };
+    };
+
+    // Unsupported, non-API function.
+    youme.read.channelAftertouch = function (channel, data) {
+        return {
+            type: "aftertouch",
+            channel: channel,
+            pressure: data[1]
+        };
+    };
+
+    // Unsupported, non-API function.
+    youme.read.twoByteValue = function (data) {
+        return (data[2] << 7) | data[1];
+    };
+
+    // Unsupported, non-API function.
+    youme.read.pitchbend = function (channel, data) {
+        return {
+            type: "pitchbend",
+            channel: channel,
+            value: youme.read.twoByteValue(data)
+        };
+    };
+
+    // Unsupported, non-API function.
+    youme.read.system = function (status, data) {
+        if (status === 1) {
+            return youme.messageFailure("quarter frame MTC");
+        }
+
+        var fn;
+        switch (status) {
+            case 0:
+                fn = youme.read.sysex;
+                break;
+            case 2:
+                fn = youme.read.songPointer;
+                break;
+            case 3:
+                fn = youme.read.songSelect;
+                break;
+            case 6:
+                fn = youme.read.tuneRequest;
+                break;
+            case 8:
+                fn = youme.read.clock;
+                break;
+            case 10:
+                fn = youme.read.start;
+                break;
+            case 11:
+                fn = youme.read["continue"];
+                break;
+            case 12:
+                fn = youme.read.stop;
+                break;
+            case 14:
+                fn = youme.read.activeSense;
+                break;
+            case 15:
+                fn = youme.read.reset;
+                break;
+            default:
+                return fluid.fail("Received an unrecognized MIDI system message: " +
+                    fluid.prettyPrintJSON(data));
+        }
+
+        return fn(data);
+    };
+
+    // Unsupported, non-API function.
+    youme.messageFailure = function (type) {
+        fluid.fail("Unsupported MIDI message type '" + type + "'.");
+    };
+
+    // Unsupported, non-API function.
+    youme.read.sysex = function (data) {
+        var begin = data[0] === 0xF0 ? 1 : 0,
```

Comment:
Explanation of the purpose of the algorithm would be helpful here, perhaps with ref to MIDI specs

Reply:
I added a typedef for the MIDI message data that included a link to the specification.  I didn't add it to every non-API function, but made sure the type was referenced in `youme.read` and `youme.write`.